yt/yt/tests/integration/queries/test_mock.py

```diff
@@ -191,3 +197,362 @@ def test_rows_limit(self, query_tracker):
+        create_user("u3")
+        create_access_control_object(
+            "aco_alter",
+            "queries",
+            attributes={
+                "principal_acl": [
+                    make_ace("allow", "u2", "administer"),
+                    make_ace("allow", "u3", "read"),
+                    make_ace("allow", "u3", "write"),
+                    make_ace("allow", "u3", "use"),
+                ]
+            })
+        q_u1 = start_query("mock", "run_forever", authenticated_user="u1", access_control_object="aco_alter")
+        q_u1.alter(authenticated_user="u2", annotations={"qwe": "asd"})
+        with raises_yt_error(AuthorizationErrorCode):
+            q_u1.alter(authenticated_user="u3", annotations={"qwe2": "asd3"})
+
+    @authors("krock21")
+    def test_alter_aco(self, query_tracker):
+        create_user("u1")
+        create_user("u2")
+        create_user("u3")
+        create_access_control_object(
+            "aco_alter_aco",
+            "queries",
+            attributes={
+                "principal_acl": [
+                    make_ace("allow", "u2", "administer"),
+                    make_ace("allow", "u3", "read"),
+                    make_ace("allow", "u3", "write"),
+                    make_ace("allow", "u3", "use"),
+                ]
+            })
+        create_access_control_object(
+            "aco_alter_aco_u3",
+            "queries",
+            attributes={
+                "principal_acl": [
+                    make_ace("allow", "u3", "administer"),
+                ]
+            })
+        q_u1 = start_query("mock", "run_forever", authenticated_user="u1", access_control_object="aco_alter_aco")
+
+        q_u1.alter(authenticated_user="u1", access_control_object="aco_alter_aco")
+
+        q_u1.alter(authenticated_user="u2", annotations={"qwe": "asd"}, access_control_object="nobody")
+        with raises_yt_error(AuthorizationErrorCode):
+            q_u1.alter(authenticated_user="u2", annotations={"qwe": "asd"}, access_control_object="aco_alter_aco")
+
+        q_u1.alter(authenticated_user="u1", access_control_object="aco_alter_aco")
+
+        with raises_yt_error(AuthorizationErrorCode):
+            q_u1.alter(authenticated_user="u3", access_control_object="nobody")
+        with raises_yt_error(AuthorizationErrorCode):
+            q_u1.alter(authenticated_user="u3", access_control_object="aco_alter_aco")
+
+        q_u1.alter(authenticated_user="u1", access_control_object="aco_alter_aco_u3")
+
+        q_u1.alter(authenticated_user="u3", access_control_object="aco_alter_aco")
+        with raises_yt_error(AuthorizationErrorCode):
+            q_u1.alter(authenticated_user="u3", access_control_object="aco_alter_aco_u3")
+
+        with raises_yt_error("Access control object \"nonexistent_aco\" does not exist"):
+            q_u1.alter(authenticated_user="u1", access_control_object="nonexistent_aco")
+
+
+# Separate list to fit 480 seconds limit for a test class.
```

Comment:
Won't NUM_TEST_PARTITIONS work here?

Reply:
I think it will, just chose to separate.
Will use NUM_TEST_PARTITIONS

Reply:
even `NUM_TEST_PARTITIONS = 32` didn't help. Preserved 2 test classes and added `NUM_TEST_PARTITIONS = 16` to each.

A test can take up to 80 seconds